qa/tasks/d4ntests.py

```diff
@@ -0,0 +1,152 @@
+                s3_user_id = 's3main'
+                log.debug(
+                    'D4N Tests: Creating user {s3_user_id}'.format(s3_user_id=s3_user_id))
+                cluster_name, daemon_type, client_id = teuthology.split_role(
+                    client)
+                client_with_id = daemon_type + '.' + client_id
+                self.ctx.cluster.only(client).run(
+                    args=[
+                        'sudo',
+                        'adjust-ulimits',
+                        'ceph-coverage',
+                        '{tdir}/archive/coverage'.format(tdir=testdir),
+                        'radosgw-admin',
+                        '-n', client_with_id,
+                        'user', 'create',
+                        '--uid', s3_user_id,
+                        '--display-name', display_name,
+                        '--access-key', access_key,
+                        '--secret', secret_key,
+                        '--email', email,
+                        '--cluster', cluster_name,
+                        ],
+                    )
+
+    def redis_startup(self):
+        try:
+            for client in self.all_clients:
+                self.ctx.cluster.only(client).run(
+                    args=[
+                        'sudo',
+                        'redis-server',
+                        '--daemonize',
+                        'yes'
+                        ],
+                    )
+    
+        except Exception as err:
+            log.debug('D4N Tests: Error starting up a Redis server')
+            log.debug(err)
+
+    def redis_shutdown(self):
+        try:
+            for client in self.all_clients:
+                self.ctx.cluster.only(client).run(
+                    args=[
+                        'sudo',
+                        'redis-cli',
+                        'shutdown',
+                        ],
+                    )
+    
+        except Exception as err:
+            log.debug('D4N Tests: Error shutting down a Redis server')
+            log.debug(err)
+
+    def remove_packages(self, client):
+        (remote,) = self.ctx.cluster.only(client).remotes.keys()
+        remove_package('s3cmd', remote)
+        remove_package('redis', remote)
+
+    def delete_user(self, client):
+        log.info("D4N Tests: Deleting S3 user...")
+        testdir = teuthology.get_testdir(self.ctx)
+
+        for user in self.user.items():
+            s3_user_id = 's3main'
+            self.ctx.cluster.only(client).run(
+                args=[
+                    'sudo',
+                    'adjust-ulimits',
+                    'ceph-coverage',
+                    '{tdir}/archive/coverage'.format(tdir=testdir),
+                    'radosgw-admin',
+                    '-n', client,
+                    'user', 'rm',
+                    '--uid', s3_user_id,
+                    '--purge-data',
+                    '--cluster', 'ceph',
+                ],
+            )
+
+task = D4NTests
```

Comment:
@samarahu I don't understand the point of this task. All I see is that a redis instance is started, an rgw user is created, and then the redis is stopped an the user is deleted. I think all of these things happen in the workunit anyway. If there isn't a point to this let's remove this file entirely.

Does this test somehow tie into what's happening in `workunints/rgw/run-d4n.sh` and `workunits/rgw/test_rgw_d4n.py`

Reply:
Those things don't happen in the workunit. The workunit assumes a redis is already running and an S3 user has already been created. I don't believe there's any reason it can't happen in the workunit, so I can add code to incorporate these steps as part of the workunit's driver script or the workunit itself but I originally wrote it this way because I was following a different suite with a similar structure. Do you think it's worth moving this setup to the workunit?

Reply:
I talked to the group about it in the standup today. We'll merge the tests as is but we'll want to make some changes around this testing. 

We thought it would be useful to have a standalone redis task that deals with starting up and shutting down redis instances like you're doing in this task. In the future I think you can move the user creation and deletion into the workunit and rename this task to redis.